### ISSUE TITLE
create-for-rbac: deprecate  '--password'

### ISSUE THREAD

**Issue by yugangw-msft** [Azure/azure-cli#7538]
Per security team's suggestion, to avoid security leak caused by simple password, we will retire this argument and only use the password auto-gen option
For users with any objections/feedback, please comment here

**Comment by ppanyukov** [comment]
Are you planning to retain `az ad sp credential reset --password` feature?  If so then this suggestions is probably OK, otherwise I have a feeling this is going to break quite a few things for many.  At least we do need to set explicit passwords in some cases using `credential reset`.

**Comment by yugangw-msft** [comment]
> Are you planning to retain az ad sp credential reset --password feature

I will retain it, for the same reason you have stated

**Comment by yugangw-msft** [comment]
@varunsh-msft, the flow exists, that users can extend an existing password's expiration period longer instead of getting a new password. The reason is the old password might have been propagated to several scripts, and updating them across comes with some risks

**Comment by varunsh-msft** [comment]
In that case, is it possible to validate that the supplied password is a GUID? If one generated the password using Azure CLI, then it will be a GUID going forward. So, if they want to extend the expiration, they can supply the same GUID back. I am concerned that by allowing any password to be specified, one can specify easy to guess passwords.

**Comment by yugangw-msft** [comment]
Yes, we can sure validate the password is strong. With that, can we apply the same check on the creating scenario, and this way we can avoid a breaking change. thoughts?

**Comment by varunsh-msft** [comment]
No no, the creating scenario must not allow choosing a password :). Let us not go backwards. We already discussed that at length. Validating strength is not as good as unique random password per service principal. People can choose easy to guess passwords, even if they are strong, and then reuse them across the place. I also shared OAuth guidance that is to generate passwords. 

I only suggested validation of GUID for reset scenario, since you wanted to allow extension of expiration.

**Comment by varunsh-msft** [comment]
Also do keep in mind that Azure Portal, PS az module, and Graph PS module do not allow choosing passwords in any of the scenarios. So, one can argue that even for reset scenario, Azure CLI should not be special, and increase risk to service principals.

**Comment by ppanyukov** [comment]
> Can you please share some scenarios where one may need to explicitly set passwords using credential reset?

@varunsh-msft : some that come to mind:

- Automated continious Azure environment provisioning.  We have scripts in bash/Azure CLI/ansible which set everything up and these can be re-run at any time. The logic of bits that deal with service principals is "create or update".  Rather than extract the auto-generated password on the first create (which is fragile), we set it to a known value. 

- We zap/recreate entire environments often, especially in development. If SP passwords change every time we do so, this will break a lot of things that need to use the SP (like web apps or VSTS release pipelines for example) and they will need to be re-configured every time.  So we set SP passwords to same values when we recreate environments.

We probably could adjust. The biggest concern with removing abiltiy to reset password to known value is that some things will definitely break, but it's very difficult to say what exactly and when. From my experiece, these things tend to break at the worst possible moment.
